cirq/ops/projectors.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Dict, List, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import value
+from cirq.ops import raw_types
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality
+class Projector(raw_types.Gate):
+    """A non-unitary gate that projects onto a single qubit."""
+
+    def __init__(self,
+                 projection_basis: List[List[float]],
+                 qid_shape: Tuple[int, ...] = (2,)):
+        """
+        Args:
+            projection_basis: a (2**num_qubits, p) matrix that lists the
+                proection vectors.
+            qid_shape: Specifies the dimension of the projection. The default is
+                2 (qubit).
+
+        Raises:
+            ValueError: If the basis vector is empty.
+        """
+        if np.prod(qid_shape) != np.array(projection_basis).shape[1]:
+            raise ValueError(
+                "Invalid shape " +
+                f"{np.array(projection_basis).shape} for qid_shape {qid_shape}")
+        self._projection_basis = projection_basis
```

Comment:
Might want to store this as a numpy array since most comparisons and logic involving it appear to be using `np.array(self._projection_basis)` right away. Thoughts ?

Reply:
Done